### ISSUE TITLE
dist.rpminspect.comparison.abidiff should not inspect programs

### ISSUE THREAD

**Issue by ppisar** [rpminspect/rpminspect#354]
dist.rpminspect.comparison.abidiff inspects programs like /usr/sbin/quota_nld:

~~~~
                "data": {
                    "item": "quota-4.04-13.el8 quota-4.04-10.el8",
                    "newnvr": "quota-4.04-13.el8",
                    "oldnvr": "quota-4.04-10.el8",
                    "scratch": false,
                    "taskid": 35990186,
                    "type": "brew_build_pair"
                },
                "outcome": "NEEDS_INSPECTION",
                "ref_url": "",
                "testcase": {
                    "name": "dist.rpminspect.comparison.abidiff",
                    "ref_url": "",
                    "test_outputs": [
                        {
                            "message": "Comparing from /usr/lib/debug/usr/sbin/quota_nld-4.04-10.el8.aarch64.debug to /usr/lib/debug/usr/sbin/quota_nld-4.04-13.el8.aarch64.debug in package quota-nld-debuginfo on aarch64 revealed ABI differences.",
                            "remedy": "ABI changes introduced during maintenance updates can lead to problems for users.  See the abidiff(1) documentation and the distribution ABI policies to determine if this detected change is allowed.",
                            "result": "NEEDS_INSPECTION",
                            "waiver_authorization": "Anyone"
                        },
~~~~

Executable programs are not shared libraries. There is no point in inspecting and reporting their ABI. Doing so only clutters the rpminspect output and increases a number of false negatives. Please stop inspecting standalone executables for ABI.

**Comment by davidmalcolm** [comment]
What about executables that expose a plugin interface?  Presumably the plugins are relying on the ABI of the program to stay consistent.

I suppose rpminspect could look for usage of "dlopen", or linkage against something that might use it.

**Comment by dcantrell** [comment]
Both good points.  How about the abidiff inspection skip ELF objects that carry a DT_INTERP value unless they also have the dlopen symbol present?  That could cover the plugin use case but skip the majority of programs.

**Comment by ppisar** [comment]
Almost any program links to glibc which dlopens network switch plugins.
I think the recursive search for dlopen wouldn't help much. 
The direct use only could be more suitable.

**Comment by dcantrell** [comment]
The idea I was thinking about was to just look at the symbols in each ELF object that rpminspect encounters.  That would not be recursive in the sense that it would not scan each linked library.  It would only examine the symbols in the current ELF object.

So in the dlopen example, it would pick up ELF executables that are built and directly use dlopen, but in most cases it would skip ELF executables.  Looking at my system here, that would check things like vim, cmake, bash, ksh93, gawk, and so on.  But it would not check /usr/sbin/quota_nld because there's no dlopen symbol present.

Thoughts?

**Comment by ppisar** [comment]
That's a good plan.

**Comment by dcantrell** [comment]
So I'm working on this one and see that the abidiff inspection was already restricted to ELF shared libraries.  The patch was contributed on 02-Sep-2020.  What version of rpminspect are you using?

Given this patch, I'm hesitant to modify it to check ELF executables that use dlopen() as the patch notes that abidiff really should be restricted to just ELF shared libraries.

The patch in question:
https://github.com/rpminspect/rpminspect/commit/53bc3b6c862e2177f4834b5178be15e95033a6bb

**Comment by ppisar** [comment]
I do not use rpminspect. I use OSCI gating which uses rpminspect. I have no
idea what version they have installed. Obviously an old one.

Which rpminspect version does contain the fix? v1.1?

**Comment by dcantrell** [comment]
Yes, v1.1 is where the fix showed up.  Current release is v1.4.  Nightly builds will have the v1.5 number.  I will check what version the OSCI gating is using.

Can you link to the full rpminspect job so I can look at the full log?

**Comment by ppisar** [comment]
https://dashboard.osci.redhat.com/#/artifact/brew-build/aid/35990186?focus=id:c14346da87d2-7
https://baseos-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/ci-test-brew-rpminspect_comparison/73191//console

I have no idea how long the links remain valid. The content usually disappears when somobody reruns the test.

**Comment by dcantrell** [comment]
I do not know how the runner is configured for this job.  It's a RHEL-8 build, but it should be using rpminspect-1.5 or later.  @msrb, @mvadkert, @jimbair ... anyone have ideas here?

**Comment by msrb** [comment]
Note rpminspect in CI now prints exactly which version is running at the very beginning:
https://osci-jenkins-1.ci.fedoraproject.org/job/fedora-ci/job/rpminspect-pipeline/job/master/20760/testReport/(root)/tests/_abidiff/

The same for downstream CI.

**Comment by dcantrell** [comment]
I'm closing this one because I feel the current rpminspect runner has taken care of this.  I cannot reproduce it locally, so I'm pretty sure it was just a matter of us cleaning up the runners.
